presto-main/src/main/java/io/prestosql/metadata/Metadata.java

```diff
@@ -398,4 +398,6 @@
     Optional<LimitApplicationResult<TableHandle>> applyLimit(Session session, TableHandle table, long limit);
 
     Optional<ConstraintApplicationResult<TableHandle>> applyFilter(Session session, TableHandle table, Constraint constraint);
+
+    Optional<TableHandle> applySample(Session session, TableHandle table, double sampleRatio);
```

Comment:
This should handle SYSTEM and BERNOULLI sampling. Some connectors (e.g., jdbc-based ones), may be able to support the more precise bernoulli sampling.

Reply:
Have updated it to pass the type of sampling along with the ratio.